### ISSUE TITLE
There's something wrong with the account switching

### ISSUE THREAD

**Issue by Kazade** [potatolondon/djangae#768]
Background info: https://p.ota.to/blog/multiple-google-account-sign-in-on-app-engine/

Now, assuming that a user has a single account, but they incorrectly click "Switch Account" on your website, the following process will happen:
1. They will be redirected to the login URL, but come straight back. Their user id will match the one before they were redirected so we will rightly assume they don't have multiple accounts
2. We log the user out, and redirect them to the login url. As they have only one account, they re-enter their password.
3. We see that their user ID matches the one before they clicked "Switch Account" and immediately log them out again.

This puts the user in a logout/login loop which they can't escape unless they browse back to the site or close the tab or whatever.

**Comment by adamalton** [comment]
Also related to this, I've just seen this in the logs for one of our projects.  It looks like maybe we need to add some kind of `login_required` check somewhere in the `switch_accounts` view:

```python
Traceback (most recent call last):
  File "/base/data/home/apps/p~my-app/21-ae5a28da0b-master.396568083451792180/sitepackages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/base/data/home/apps/p~my-app/21-ae5a28da0b-master.396568083451792180/sitepackages/djangae/contrib/gauth/views.py", line 43, in switch_accounts
    user_hash = hashlib.sha1(current_google_user.user_id()).hexdigest()
AttributeError: 'NoneType' object has no attribute 'user_id'
```

**Comment by Kazade** [comment]
This was fixed recently in #847
